### PR TITLE
Add support for delivery service thresholds in goTM

### PR DESCRIPTION
**Pull request by dneuman64**
goTM now checks the Total TPS Threshold and Total Kbps Threshold defined on a delivery service and sets a delivery service to available or unavailable accordingly.  Also made sure Total TPS was properly calculated and displayed in the DS stats API as well as  ensuring delivery service state was correctly reflected in the UI.

### CODE REVIEW

File: traffic_monitor/experimental/traffic_monitor/deliveryservice/stat.go [hunk 1]
```diff
@@ -72,7 +74,7 @@ func setStaticData(dsStats Stats, dsServers map[enum.DeliveryServiceName][]enum.
 	return dsStats
 }
 
-func addAvailableData(dsStats Stats, crStates peer.Crstates, serverCachegroups map[enum.CacheName]enum.CacheGroupName, serverDs map[enum.CacheName][]enum.DeliveryServiceName, serverTypes map[enum.CacheName]enum.CacheType, precomputed map[enum.CacheName]cache.PrecomputedData) (Stats, error) {
+func addAvailableData(dsStats Stats, crStates peer.Crstates, serverCachegroups map[enum.CacheName]enum.CacheGroupName, serverDs map[enum.CacheName][]enum.DeliveryServiceName, serverTypes map[enum.CacheName]enum.CacheType, precomputed map[enum.CacheName]cache.PrecomputedData, mc to.TrafficMonitorConfigMap) (Stats, error) {
```
Comment (rob05c) [inline traffic_monitor/experimental/traffic_monitor/deliveryservice/stat.go]: Looks like `mc` isn't used anymore, can it be removed?

File: traffic_monitor/experimental/traffic_monitor/static/index.html [hunk 2]
```diff
@@ -382,7 +386,7 @@
 
 					 // \todo check that array has a member before dereferencing [0]
 					 if (jds[deliveryService].hasOwnProperty("isAvailable")) {
-						 document.getElementById("deliveryservice-stats-" + deliveryService + "-status").textContent = jds[deliveryService]["isAvailable"][0].value ? "available" : "unavailable";
+						 document.getElementById("deliveryservice-stats-" + deliveryService + "-status").textContent = jds[deliveryService]["isAvailable"][0].value == "true" ? " available" : "unavailable - " + jds[deliveryService]["error-string"][0].value;
```
Comment (rob05c) [inline traffic_monitor/experimental/traffic_monitor/static/index.html]: Is the space before " available" intentional? If so, should "unavailable" also have a space before it?

**Comment by rob05c** [comment]
Other than those 2 little things, looks great. Not enough to -1, if they take more than 5mins.

File: traffic_monitor/experimental/traffic_monitor/deliveryservice/stat.go (hunk 1 above)
Comment (dneuman64) [inline traffic_monitor/experimental/traffic_monitor/deliveryservice/stat.go]: ah yes, I missed it

**Comment by dneuman64** [comment]
@rob05c fixed

**Pull request closed without merging**